src/runners/scilla_runner.ml

```diff
@@ -278,22 +297,48 @@ let () =
         in
         let m = Msg mmsg in
 
-        (* Retrieve state variables *)
-        let (curargs, cur_bal) = 
-        try
-          input_state_json cli.input_state
-        with
-        | Invalid_json s ->
-            fatal_error_gas
-              (s @ (mk_error0 (sprintf "Failed to parse json %s:\n" cli.input_state)))
-            gas_remaining
+        let cstate, gas_remaining' = 
+        if is_ipc then
```

Comment:
So, we retrieve the state in two different ways, right? Can we make it more uniform, so the initialisation from the local state and full input state would follow the same interface?

Reply:
>So, we retrieve the state in two different ways, right? 
Yes.

>Can we make it more uniform, so the initialisation from the local state and full input state would follow the same interface?

I actually spent quite a while doing that, but unfortunately, wasn't successful :-(. The requirements (at the executable level) vary b/w the two modes and also b/w whether deployment or not. So this seemed like the only way to do it.